source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -61,6 +61,9 @@ void ExtProcLoggingInfo::recordGrpcCall(
     ProcessorState::CallbackState callback_state,
     envoy::config::core::v3::TrafficDirection traffic_direction) {
   ASSERT(callback_state != ProcessorState::CallbackState::Idle);
+  if (grpcCalls(traffic_direction).size() >= kMaxGrpcLogging) {
```

Comment:
This probably solved the issue we saw. But it seems that we will always `pop one and push one` once we reach the the boundary of `kMaxGrpcLogging` , which is not ideal for performance.

So I am thinking should we think about a more aggressive removal like remove first N elements so that we don't need to pop_front every time once it reaches boundary.